homeassistant/components/light/tcpbulbs.py

```diff
@@ -0,0 +1,146 @@
+"""
+Support for TCP Connected (Greenwave Reality) lights.
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.tcpbulbs/
+"""
+import logging
+import voluptuous as vol
+# Import the device class from the component that you want to support
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, Light, PLATFORM_SCHEMA, SUPPORT_BRIGHTNESS)
+from homeassistant.const import CONF_HOST
+import homeassistant.helpers.config_validation as cv
+# Home Assistant depends on 3rd party packages for API specific code.
+SUPPORTED_FEATURES = (SUPPORT_BRIGHTNESS)
+_LOGGER = logging.getLogger(__name__)
+# Validation of the user's configuration
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+})
+
+
+def pullXML(host, bulbxml):
+    REQUIREMENTS = ['xmltodict']
+    import xmltodict
+    resp = tcpAction(host, 0, 'xml', 0)
+    # Store bulb information as xml file
+    with open(bulbxml, 'wb') as f:
+        f.write(resp.content)
+
+    with open(bulbxml) as fd:
+        doc = xmltodict.parse(fd.read())
+        return doc
+
+def numMap(value, leftMin, leftMax, rightMin, rightMax):
+    # Figure out how 'wide' each range is
+    leftSpan = leftMax - leftMin
+    rightSpan = rightMax - rightMin
+
+    # Convert the left range into a 0-1 range (float)
+    valueScaled = float(value - leftMin) / float(leftSpan)
+
+    # Convert the 0-1 range into a value in the right range.
+    return int(rightMin + (valueScaled * rightSpan))
+
+def getBrightness(device):
+        if 'level' in device:
+            level = numMap(int(device['level']),1,100,1,255)
+            return(level)
+        else:
+            return(0)
+            
+def checkOnline(device):
+        if 'offline' in device:
+            return(False)
+        else:
+            return(True)

```

Comment:
I think `is_online(dev)` would be a nicer. Maybe something like:
```
def is_online(device):
    return 'offline' not in device
```

Reply:
This function checks if the offline tag exists for the device, if it exists the device is offline. Not sure if your code will do that

Reply:
It will :)

```
if X:
  return False
else:
  return True
```

can be simplified to just:

```
return not X
```